google/cloud/pubsub/internal/streaming_subscription_batch_source.cc

```diff
@@ -236,11 +236,16 @@ void StreamingSubscriptionBatchSource::ExtendLeases(
   }
   lk.unlock();
   for (auto& r : split) {
+    std::unique_lock<std::mutex> lk(mu_);
+    nonce_++;
+    lk.unlock();
+    callback_->StartModackSpan(request, nonce_);
```

Comment:
You can't read `nonce_` outside the lock.

Reply:
You're right. Fixed.

Reply:
Typically we would write this just as ...
```
  lk.lock();
  auto nonce = nonce_++;
  lk.unlock();
```
to minimize the extent of the critical region.

[Aside: I would try to pick a better name, like, say, `modack_id`, to match the others, even if it is client generated.]

Reply:
That makes more sense, thanks Brad. Done, also renamed.